src/components/structural/header/Header.js

```diff
@@ -198,38 +179,55 @@ class Header extends Component {
-                            open={this.state.logMenuOpen}
-                            onClick={() => this.setState({ logMenuOpen: !this.state.logMenuOpen })}
-                            label="logout"
-                            style={{ marginTop: 5 }} />
-                        <Popover
-                            open={this.state.logMenuOpen}
-                            anchorEl={document.getElementById("user")}
-                            anchorOrigin={{ horizontal: "right", vertical: "bottom" }}
-                            onClose={this.handleLogClick} >
-                            <MenuItem primarytext="Log Out" onClick={this.logout} >Log Out</MenuItem>
-                        </Popover>
-                    </Fragment>
+                {this.props.user && this.props.user.name ?
+                    <GoogleLogout
+                        clientId="12320757914-puvjd9tbs79sc19sl433gqp9sci1tfqt.apps.googleusercontent.com"
+                        buttonText="Logout"
+                        render={renderProps => (
+                            <Fragment>
+                                <Avatar
+                                    id="login"
+                                    src={this.props.user.imageUrl}
+                                    open={this.state.logMenuOpen}
+                                    onClick={() => this.setState({ logMenuOpen: !this.state.logMenuOpen })}
+                                    label="logout"
+                                    style={{ marginTop: 5 }} />
+                                <Popover
+                                    open={this.state.logMenuOpen}
+                                    anchorEl={document.getElementById("user")}
+                                    anchorOrigin={{ horizontal: "right", vertical: "bottom" }}
+                                    onClose={this.handleLogClick} >
+                                    <MenuItem primarytext="Log Out" onClick={renderProps.onClick}>Log Out</MenuItem>
+                                </Popover>
+                            </Fragment>
+                        )}
+                        onLogoutSuccess={this.logout}
+                        onFailure={(err) => console.error("Could not logout: ", err) }
+                    />
                     :
-                    <Button
-                        type="button"
-                        variant="outlined"
-                        size="small"
-                        color="primary"
-                        onClick={this.login}
-                        style={{
-                            color: "white",
-                            margin: 4,
-                            padding: 2,
-                            border: "1px solid #fff"
-                        }}>
-                        Log In
-                    </Button>
+                    <GoogleLogin
+                        clientId="12320757914-puvjd9tbs79sc19sl433gqp9sci1tfqt.apps.googleusercontent.com"
```

Comment:
We should keep this out of source control and use some form of environment variable for it.